assets/src/ba_data/python/bastd/game/easteregghunt.py

```diff
@@ -186,9 +186,13 @@ def _update(self) -> None:
         xpos = random.uniform(-7.1, 6.0)
         ypos = random.uniform(3.5, 3.5)
         zpos = random.uniform(-8.2, 3.7)
+        def _is_exists(egg: Egg) -> bool:
+            if egg.node is None:
+                return False
+            return egg.node.exists()
 
         # Prune dead eggs from our list.
-        self._eggs = [e for e in self._eggs if e]
+        self._eggs = [e for e in self._eggs if _is_exists(e)]
```

Comment:
Interesting; the original code is supposed to be doing the same thing as this; the .exists() method for the egg should return false when the node is None or an invalid reference, and `if actor` now should be the same as `if actor.exists()`. I'll take a look and see which of that is not happening...